### PR TITLE
Haohui - Gives the user permission to manage individual user permissions. 

### PR DESCRIPTION
**Pull request by Lin1404**
# Description
Please include the exact bug/functionality description and a summary of the changes/ related issues. Please also include any other relevant motivation and context:
image

## Related PRS (if any):
This frontend PR is related to the backend PR # 532.
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as owner user call user A
4. go to dashboard→ Other Links→ Permissions Management→ Manage User Permissions
5. input another user name, call user B
6. find the "Manage User Permissions", click the add button
7. scroll down and click submit button
8. log out and login as user B
9. go to dashboard→ Other Links→ Permissions Management→ Manage User Permissions
10. check the user B able to manage user permissions.

## Screenshots or videos of changes:
https://drive.google.com/file/d/1gEo7ixKdH1-o3ZnFts0N_L_Hsm73m34M/view?usp=sharing

## Note:
Include the information the reviewers need to know.

### CODE REVIEW

File: src/components/PermissionsManagement/UserPermissionsPopUp.jsx [hunk 1]
```diff
@@ -43,6 +43,7 @@ const UserPermissionsPopUp = ({ allUserProfiles, toggle, getAllUsers, roles }) =
       const newPermissionsObject = {
         frontPermissions: actualPermissionsFront,
       };
+      console.log(newPermissionsObject.frontPermissions)
```
Comment (GaryB93) [inline src/components/PermissionsManagement/UserPermissionsPopUp.jsx:46]: @Lin1404 Can we remove the console.log?

**Review by GaryB93 — DISMISSED** [review]
Great work on implementing the new permission functionality. I was able to add the Manage User Permissions on a Volunteer role and that user account was only allowed to manage user permissions on the User Roles page and was restricted from the rest of the functionality on that page.
Screenshot 2023-09-11 at 3 20 11 PM

**Review by robertoooc — APPROVED** [review]
Hi @Lin1404, I tested your pr along with the backend implementation as well and it performed as intended. Didn't catch any issues while testing it as well. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/117053202/718b3ef0-4670-4e46-ba97-f393aeb4deff

**Comment by AriaYu927** [comment]
Hi @Lin1404 , I've tested your PR and it works as expected. Thank you!

Before added permission to YIhantest Core:
截屏2023-09-13 13 25 33

After added permission:
截屏2023-09-13 13 26 43

File: src/components/PermissionsManagement/UserPermissionsPopUp.jsx (hunk 1 above)
Comment (Lin1404) [reply src/components/PermissionsManagement/UserPermissionsPopUp.jsx]: Good catch! Thank you Gary!

File: src/routes.js [hunk 2]
```diff
@@ -123,7 +123,7 @@ export default (
             path="/permissionsmanagement"
             exact
             component={PermissionsManagement}
-            routePermissions={RoutePermissions.permissionsManagement}
+            routePermissions={[RoutePermissions.permissionsManagement,RoutePermissions.userPermissionsrManagement]}
```
Comment (GaryB93) [inline src/routes.js:126]: Hey @Lin1404, is there supposed to be a 'r' between Permissions and Management on `RoutePermissions.userPermissionsrManagement` ?

**Review by GaryB93 — DISMISSED** [review]
Hey @Lin1404, tested this updated PR with backend PR#532 to verify the permission of Manage User Permissions can be given to an otherwise unauthorized user and would function as expected.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108501006/abde0258-7ca5-4f5a-a3c3-398f00d378d9

**Review by AriaYu927 — DISMISSED** [review]

**Comment by wantingxu7** [comment]
Hi @Lin1404 , your work works as expected in my local machine, good job!
User B_before given permission
f988e)
User B_after given permission_p2

**Review by wantingxu7 — APPROVED** [review]

**Review by Shiwani99 — CHANGES_REQUESTED** [review]
Hi @Lin1404,
I tested your PR and I wanted to add a problem that I faced. When I set user management permissions from owner to manager I'm able to see user management in manager, but when I set permission for volunteer from manager, I'm not able to see the permissions get saved after I hit submit in manager profile because of which I'm not able to see other links in volunteer page.  This is also happening for one other volunteer profile I created where in I removed my user management permission from manager to another volunteer but it is not getting removed despite giving submit. I tried setting permissions for test account too from manager, but I'm encountering the same issue. There is no issue when I try to add or remove user management permission for any user form owner account.
I'm attaching a screen recording of what is happening. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/c9bf0365-f67b-4828-9e1e-6edb2d206ac0

**Review by DavidC0126 — APPROVED** [review]
Hi @Lin1404 I tested your PR and it works as described.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/74082492/312c67b3-6908-4c95-9aa7-1f7f69e6f608

**Comment by Lin1404** [comment]
> Hi @Lin1404 I tested your PR and it works as described.
> 
>  permission_user.mov

Hi @DavidC0126 I retested it and I think the problem is fixed now.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78375725/61f0182e-b9a5-48ca-9980-fdb51290efd5

**Review by Jiyuanxia — APPROVED** [review]
Hi @Lin1404 , I have reviewed your pull request and tested it. It works as you intended. Well done!
Screenshot 2023-09-21 222357
Screenshot 2023-09-21 222217

**Review by harshidad — APPROVED** [review]
I tested the PR with backend PR#532 to verify the permission of users and it's working as expected. Great Work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/109878936/23d0063b-7ce8-46b1-aa97-81b632066bba

**Review by belagaam — APPROVED** [review]
I tested the PR as per the instructions and everything worked as expected.




















https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47104198/6681a12e-2c23-4ccf-a1c1-4a8e34c4071c

**Review by EdwinLauM — APPROVED** [review]
Hey @Lin1404! Works as expected, good job! 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/42288987/900359c5-5317-45d0-8a70-c3fc3db73e39

**Comment by Lin1404** [comment]
> Hi @Lin1404, I tested your PR and I wanted to add a problem that I faced. When I set user management permissions from owner to manager I'm able to see user management in manager, but when I set permission for volunteer from manager, I'm not able to see the permissions get saved after I hit submit in manager profile because of which I'm not able to see other links in volunteer page. This is also happening for one other volunteer profile I created where in I removed my user management permission from manager to another volunteer but it is not getting removed despite giving submit. I tried setting permissions for test account too from manager, but I'm encountering the same issue. There is no issue when I try to add or remove user management permission for any user from owner account. I'm attaching a screen recording of what is happening.
> 
>  Screen.Recording.2023-09-14.at.6.01.38.PM.mov

Hi @Shiwani99 , the reason the bug happen because the permission type different. I have fixed and retest it, I believe it works well now. Please check out the retest video below.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78375725/61f0182e-b9a5-48ca-9980-fdb51290efd5

**Review by Shiwani99 — APPROVED** [review]
Hi,
Thank you for the changes. Great work. I retested it and the bug was resolved now. It works as expected. I'm attaching screen recording of my test.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/4f6f1ebc-e94e-4813-ba05-ee272c1cff71

**Comment by one-community** [comment]
Thank you all, moving to final review

**Comment by emunkh404** [comment]
Nice work

**Pull request merged**